packages/lib/passport-interface/src/PODPipeline/inputToPODEntries.ts

```diff
@@ -0,0 +1,116 @@
+import { PODValue } from "@pcd/pod";
+import { uuidToBigInt } from "@pcd/util";
+import {
+  PODPipelineInputFieldType,
+  PODPipelinePODEntry
+} from "../genericIssuanceTypes";
+import { FieldTypeToJavaScriptType } from "./Input";
+
+/**
+ * Conversions for converting input values to POD values.
+ * If no valid conversion exists, the conversion is set to undefined.
+ * This conversion happens *after* the raw values from the CSV file are
+ * coerced to the correct type by the CSVInput class.
+ *
+ * This maps from supported POD value types (not all value types are supported
+ * yet) to the input types that can be converted to them. Input types are
+ * described by {@link PODPipelineInputFieldType}.
+ */
+type SupportedPODValueType = Exclude<PODValue["type"], "eddsa_pubkey">;
+
+type Conversions = {
+  [P in SupportedPODValueType]: {
+    [T in PODPipelineInputFieldType]?: (
+      input: FieldTypeToJavaScriptType<T>
+    ) => PODValue;
+  };
+};
+
+const PODValueConversions: Conversions = {
```

Comment:
I still feel this way, though I might be missing the big picture of the design here.

Reply:
The question is whether there's any good case for being able to choose the POD value type, having already chosen the input field type. It seems we could have the following mappings with no choice for the user:

String -> String
Int -> Int
Cryptographic -> Cryptographic
EdDSA Public Key -> EdDSA Public Key
Date -> Int
Boolean -> Int
UUID -> String

Do these seem correct?

Reply:
My current thinking for UUIDs (which will be implemented as a first-class type in PODs soon) is that UUID should map to a number (cryptographic for now, uuid type in future), by extracting its binary representation.  If we're going to enforce that a UUID is well-formed then making it numeric makes most sense to me.  If that's not enforced, then it may as well be a string.  In TS/JS it's a bit of a toss-up, but in other languages I'd expect native UUID types to be more strict and int-based rather than string-based.

Otherwise those mappings seem correct.  Importantly, all the types you list are types I intend to add as PODValue types.  In general, do you see a reason for there to ever be types which don't map automatically to specific PODValue types?  I think the extra layer of translation is probably unnecessary, and can be hidden from the user.  If a user wanted to do something unusual, like represent an integer as a string, they could enter as the string "12345".

Reply:
Interestingly, my mental model has me thinking of UUID->Int as simpler/cheaper, which I realize is not entirely accurate given that Poseidon hashes turn out to be more expensive than SHA256 hashes (as discussed elsewhere).  I still think UUID->int is probably the right call, since in general it means there are more things we could choose to do in a circuit with the value (e.g. prove it's a valid UUID, or has a specific prefix or format).  I don't think we need any of those things now, but for minimal design cost it seems best to maintain the possibility.